pkg/sshd/circuit.go

```diff
@@ -0,0 +1,61 @@
+package sshd
+
+import (
+	"fmt"
+	"sync/atomic"
+)
+
+// CircuitState represents the state of a Circuit
+type CircuitState uint32
```

Comment:
Maybe I'm not understanding the lower level synchronization types here, but from reading up to understand this bit of code from sync/atomic's docs:

> These functions require great care to be used correctly. Except for special, low-level applications, synchronization is better done with channels or the facilities of the sync package. Share memory by communicating; don't communicate by sharing memory.

Not saying this should be refactored (this looks good to me), but I'm curious on your decision to go this route instead of using a combination of a boolean and mutal exclusion locks instead. Looking at the code, it looks like we only need two circuit states: open and closed.